tests/console/hwloc_testsuite.pm

```diff
@@ -0,0 +1,36 @@
+# SUSE's hwloc tests
+#
+# Copyright © 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Run hwloc testsuite
+# Maintainer: Thomas Blume <tblume@suse.com>
+
+use base "consoletest";
+use warnings;
+use strict;
+use testapi;
+
+sub run() {
+    # install hwloc testsuite
+    script_run('zypper ar http://download.suse.de/ibs/QA:/SLE12SP2/update/ hwloc-testrepo');
```

Comment:
And a hint for the general procedure. Try to be very restrictive regarding the test results and exit codes of intermediate steps, i.e. use `assert_script_run` in place of `script_run` unless you want to explicitly check the return value of a command. `utils::zypper_call` as suggested by @Soulofdestiny is doing that internally already.

Regarding the use of versions in external dependencies. Why should we add a SLE12-SP2 repo to SLE12-SP3 and also SLE15 tests? Maybe we can ensure there is a version specific repository corresponding to the variable content of `VERSION`?